src/Module.ts

```diff
@@ -590,12 +590,32 @@ export default class Module {
 		return this.needsTreeshakingPass;
 	}
 
-	getAndCreateNamespace(): NamespaceVariable {
-		if (!this.declarations['*']) {
-			this.declarations['*'] = new NamespaceVariable(this.astContext);
+	getOrCreateNamespace(): NamespaceVariable {
+		const namespace = this.declarations['*'];
+		if (namespace) return namespace;
+
+		return (this.declarations['*'] = new NamespaceVariable(this.astContext));
+	}
+
+	private includeNamespace() {
+		const namespace = this.getOrCreateNamespace();
+		if (namespace.needsNamespaceBlock) return;
+
+		let hasReexports = false;
+		for (const importName in this.reexports) {
+			if (importName in this.exports) continue;
```

Comment:
Removing this line does not produce any red tests. What is the condition in which this can happen and can we add this to a test?

Reply:
You're completely right - this check is unnecessary as exports shadowing reexports throw an early error.

Reply:
👍